pkg/hnsw/friends.go

```diff
@@ -63,6 +65,45 @@ func (v *Friends) GetFriendsAtLevel(level int) (*DistHeap, error) {
 	return v.friends[level], nil
 }
 
+func (v *Friends) Flush(numNeighbors int) []byte {
+	if len(v.friends) == 0 {
+		panic("no levels to be found")
+	}
+
+	// for every neighbor, we're going to serialize
+	// +-------+-------------------------+
+	// | level |      Id
+
+	buf := make([]byte, (4+1)*numNeighbors)
+
+	level0 := v.friends[0]
+	copyLevel0 := level0.Clone()
+
+	for i := 0; i < numNeighbors; i++ {
+		if copyLevel0.IsEmpty() {
+			// write out max values here
+			continue
+		}
+
+		closestItem, err := copyLevel0.PopMinItem()
+		if err != nil {
+			panic(fmt.Sprintf("failed to find closest item in friends: %v", err))
```

Comment:
Return the err instead of panicking, it tends to be pretty annoying to work around this later.